<a href="https://colab.research.google.com/github/TEERAWATL/Market-basket-analysis/blob/main/Market_basket_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install apyori

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Preparing metadata (setup.py) ... done
  Created wheel for apyori: filename=apyori-1.1.2-py3-none-any.whl size=5976 sha256=7ff2d2ddf0ebf169296c0e413b011edc250234fe107d96b0b1b9f93a8de72dff
  Stored in directory: /root/.cache/pip/wheels/32/2a/54/10c595515f385f3726642b10c60bf788029e8f3a1323e3913a
Successfully built apyori


In [2]:
import pandas as pd
from apyori import apriori

In [3]:
transactions = [
    ["eggs", "bread", "milk"],
    ["eggs", "bread", "butter"],
    ["eggs", "coffee", "bread"],
    ["bread", "milk", "butter"],
    ["coffee", "milk", "sugar"],
    ["bread", "eggs", "milk", "coffee"],
    ["eggs", "bread", "milk", "butter"],
]

In [8]:
min_support = 0.3
min_confidence = 0.5
min_lift = 1.0
min_length = 2

In [9]:
rules = apriori(transactions,
                min_support=min_support,
                min_confidence=min_confidence,
                min_lift=min_lift,
                min_length=min_length)


In [10]:
rules_df = pd.DataFrame(
    [
        {
            "items_base": frozenset(r.ordered_statistics[0].items_base),
            "items_add": frozenset(r.ordered_statistics[0].items_add),
            "support": r.support,
            "confidence": r.ordered_statistics[0].confidence,
            "lift": r.ordered_statistics[0].lift,
        }
        for r in rules
    ]
)


In [11]:
print(rules_df)


  items_base      items_add   support  confidence      lift
0         ()        (bread)  0.857143    0.857143  1.000000
1         ()         (eggs)  0.714286    0.714286  1.000000
2         ()         (milk)  0.714286    0.714286  1.000000
3    (bread)       (butter)  0.428571    0.500000  1.166667
4         ()  (eggs, bread)  0.714286    0.714286  1.000000
5         ()  (milk, bread)  0.571429    0.571429  1.000000
6    (bread)   (eggs, milk)  0.428571    0.500000  1.166667


**For example**

items_base: (eggs) - The antecedent, or the item(s) on the left-hand side of the association rule.

items_add: (bread) - The consequent, or the item(s) on the right-hand side of the association rule.

support: 0.714286 - The proportion of transactions that contain both the antecedent and the consequent items. In this case, the support value means that 71.43% of transactions contain both eggs and bread.

confidence: 0.833333 - The probability of finding the consequent (bread) in a transaction, given that the antecedent (eggs) is present. In this case, the confidence value means that 83.33% of transactions that contain eggs also contain bread.

lift: 1.666667 - The ratio of observed support to expected support if the antecedent and consequent items were independent. A lift greater than 1 indicates that the antecedent and consequent items are more likely to be purchased together than individually. In this case, the lift value means that eggs and bread are 1.67 times more likely to be purchased together than if they were purchased independently.